### PR TITLE
`Enpass` as Flatpak -- io.enpass.Enpass

### PR DESCRIPTION
**Pull request by ShobuPrime**
### Please confirm your submission meets all the criteria

- [:heavy_check_mark:] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [:heavy_check_mark:] My pull request follows the instructions at [App Submission][submission].
- [:heavy_check_mark:] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
```
finish-args:
    # X11 + XShm access
  - --share=ipc
  - --socket=x11
    # Notification access
  - --talk-name=org.freedesktop.Notifications
    # System Tray Icon
  - --talk-name=com.kde.StatusNotifierWatcher
    # Secret Service
  - --own-name=org.freedesktop.secrets
    # Favicon download
  - --share=network
    # SSH Agent access
  - --socket=ssh-auth
    # Required to save vaults in default location <$HOME/Documents>
  - --filesystem=home
  ```
- [:heavy_check_mark:] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [:no_entry:] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://discussion.enpass.io/index.php?/topic/8076-linux-make-an-official-enpass-appimagesnapflatpak/&do=findComment&comment=32290**
- [:no_entry:] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [:no_entry:] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* -- `There are no patches, only wrapping Debian package as Flatpak`

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission
---
Application verified to build and run locally on machine

### CODE REVIEW

**Comment by ShobuPrime** [comment]
Community Package to solve this request: [[Linux] Make an Official Enpass AppImage/Snap/Flatpak](https://discussion.enpass.io/index.php?/topic/8076-linux-make-an-official-enpass-appimagesnapflatpak/) for [Steam Deck](https://www.steamdeck.com/en/) users

**Comment by ShobuPrime** [comment]
bot, build io.enpass.Enpass

**Comment by ShobuPrime** [comment]
bot, build io.enpass.Enpass

**Comment by ShobuPrime** [comment]
bot, build io.enpass.Enpass

File: io.enpass.Enpass.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# io.enpass.Enpass.yaml
+app-id: io.enpass.Enpass
+default-branch: stable
```
Comment (hfiguiere) [inline io.enpass.Enpass.yaml:3]:
```suggestion
```

File: io.enpass.Enpass.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            bsdtar -xvf enpass.deb
+            tar -xzf data.tar.gz -C deb-package
+
+            # Install desktop entry and icons
+            mv deb-package/usr/share/{applications,icons} export/share/
+
+            rename enpass ${FLATPAK_ID} export/share/applications/*.desktop
+            rename enpass ${FLATPAK_ID} export/share/icons/hicolor/*/{apps,status}/*
+
+            sed -i "s/Exec=\/opt\/enpass\/Enpass %U/Exec=\/app\/extra\/enpass\/Enpass %F/g" export/share/applications/${FLATPAK_ID}.desktop
```
Comment (hfiguiere) [inline io.enpass.Enpass.yaml:75]: 1. you should use `destkop-file-edit`
2. the executable should NOT be an absolute path. Instead it should match the `command` property, modulo the arguments. `enpass %F`

Same for the next two lines.

**Comment by hfiguiere** [comment]
build fail because you never install the icon.

**Comment by hfiguiere** [comment]
also you'll likely need a desktop file to export that in not part of the extra-data.

File: io.enpass.Enpass.yaml (hunk 2 above)
Comment (ShobuPrime) [reply io.enpass.Enpass.yaml:75]: Thanks for the suggestions @hfiguiere 

- `desktop-file-edit` was failing when trying to build locally on an Arch Linux VM, saying it couldn't be found. Consequently, I went the `sed` route.
- Will `${FLATPAK_DEST}/extra/enpass/Enpass` suffice throughout the manifest?

**Comment by ShobuPrime** [comment]
bot, build io.enpass.Enpass

**Comment by ShobuPrime** [comment]
bot, build io.enpass.Enpass

**Comment by ShobuPrime** [comment]
bot, build io.enpass.Enpass

File: io.enpass.Enpass.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+        install -Dm 755 apply_extra --target-directory ${FLATPAK_DEST}/bin/
+        chmod +x ${FLATPAK_DEST}/bin/apply_extra
+
+        install -Dm 755 enpass --target-directory ${FLATPAK_DEST}/bin/
+        chmod +x ${FLATPAK_DEST}/bin/enpass
+
+        install -Dm 644 ${FLATPAK_ID}.metainfo.xml --target-directory ${FLATPAK_DEST}/share/metainfo
+
+        for size in 16 24 32 48 64 96 128 256; do
```
Comment (hfiguiere) [inline io.enpass.Enpass.yaml:41]: you only need 128 or 256. Anything else is superfluous.

File: io.enpass.Enpass.yaml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+          install -Dm 644 enpass_${size}.png ${FLATPAK_DEST}/share/app-info/icons/flatpak/${size}x${size}/${FLATPAK_ID}.png
+          install -Dm 644 enpass_${size}.png ${FLATPAK_DEST}/share/icons/hicolor/${size}x${size}/apps/${FLATPAK_ID}.png
+          rm enpass_${size}.png
```
Comment (hfiguiere) [inline io.enpass.Enpass.yaml:44]:
```suggestion
```

**Review by hfiguiere — COMMENTED** [review]
What about that logo.png? You never use the file.

File: io.enpass.Enpass.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    # System Tray Icon (Test at every major update)
+  - --talk-name=com.kde.StatusNotifierWatcher
+    # Secret Service
+  - --own-name=org.freedesktop.secrets
+    # Favicon download
+  - --share=network
+    # SSH Agent access
+  - --socket=ssh-auth
+    # Required to save vaults in default location <$HOME/Documents>
+  - --filesystem=home
```
Comment (hfiguiere) [inline io.enpass.Enpass.yaml:24]: then xdg-documents.

or `persist`.

File: io.enpass.Enpass.yaml (hunk 2 above)
Comment (ShobuPrime) [reply io.enpass.Enpass.yaml:81]: Hey @hfiguiere, the bot has successfully built my manifest, and I've verified I can install and run the Flatpak with `flatpak install --user https://dl.flathub.org/build-repo/9396/io.enpass.Enpass.flatpakref`

I'm still only able to get the Flatpak to work with absolute paths for the exec. Variables such as `FLATPACK_DEST`, `FLATPAK_SYSTEM_DIR`, `XDG_RUNTIME_DIR` as prefixes would not run the app.

In addition, `desktop-file-edit` would not work when trying to build locally (even when verified at /usr/bin/desktop-file-edit)` on my builder.

File: io.enpass.Enpass.yaml (hunk 5 above)
Comment (ShobuPrime) [reply io.enpass.Enpass.yaml:24]: [I originally did](https://github.com/ShobuPrime/flathub/commit/826ce86bc219e317216ff93fd5fc4da2e6fbc4a6) use `xdg-documents`, but the [bot didn't like that](https://buildbot.flathub.org/#/builders/37/builds/3299)

Comment (JakobDev) [reply io.enpass.Enpass.yaml:24]: You just need to add it to [this list](https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json)

Comment (hfiguiere) [reply io.enpass.Enpass.yaml:24]: neither.

The bot is fine with `xdg-documents`. its `xdg-data` and `xdg-config` it doesn't like

**Comment by ShobuPrime** [comment]
Rebuilding Flatpak with new recommendations, and will create new PR when builds are successful again

**Comment by hfiguiere** [comment]
you can just wait and update the PR to not make life harder on reviewers.

**Pull request closed without merging**